flask_monitoringdashboard/core/cache.py

```diff
@@ -64,29 +65,60 @@ def init_cache():
             )
 
 
+def add_to_cache(endpoint_name):
+    """
+        This should be added to all endpoint no directly hosted in Flask before the first request.
+        It initializes the in-memory cache from the db
+    """
+    global memory_cache
+    if not memory_cache.get(endpoint_name):
+        with DatabaseConnectionWrapper().database_connection.session_scope() as session:
+            last_req_dict = dict(get_last_requested(session))
+            hits_dict = dict(get_endpoints_hits(session))
+            averages_dict = dict(get_endpoint_averages(session))
+            memory_cache[endpoint_name] = EndpointInfo(
+                last_requested=last_req_dict.get(endpoint_name),
+                average_duration=averages_dict.get(endpoint_name),
+                hits=hits_dict.get(endpoint_name),
+            )
+
+
 def update_last_requested_cache(endpoint_name):
     """
     Use this instead of updating the last requested to the database.
     """
     global memory_cache
-    memory_cache.get(endpoint_name).set_last_requested(datetime.datetime.utcnow())
+    try:
+        memory_cache.get(endpoint_name).set_last_requested(datetime.datetime.utcnow())
+    except Exception as error:
```

Comment:
Don't we have more detailed Exception info? Why are we catching Exception?

Reply:
I was having a TypeException because the memory cache was not initialized. The reason for that is that I enabled my MongoDB cluster to read from the secondary which guarantees only eventual consistency of data. Do you want me to be more specific with the exception of being caught?